### DISCUSSION TITLE
TemporaryUploadedFile and SplFileInfo properties aren't synched

### DISCUSSION THREAD

**Discussion by JulioPablo** [livewire/livewire#3055] [7. Bugs (archived)]
### Description
I'm trying to rename a file upon upload to restore the file extension, this is not a new issue and not even really Livewire related, files of certain extensions getting them changed, but now that I'm trying to manually restore the files to their original extension I've noticed that the `SplFileInfo` stuff inherited by the `TemporaryUploadedFile`, seems to not be synched, the real filename & path seem to live in the `TemporaryUploadedFile` `path` property and then the `SplFileInfo` properties/methods return completely different paths & names. So when I try to rename the file it fails.

![image](https://user-images.githubusercontent.com/33587721/112199783-22e9b000-8bd4-11eb-954f-2b28125eec0e.png)

Perhaps I'm missing something or perhaps this unexpected behavior, please do let me know.
Regards.

**Context**
- Livewire version: 2.2.3
- Laravel version: 8.0
- Alpine version: 2.3.0
- Running in php:8-apache docker image.

**Comment by nuernbergerA** [comment]
Hey @JulioPablo,

you have to "store" the uploaded file like so
```php
$this->file->storeAs('path', 'file.txt', 'optional-disk');
```

if you want to see the original filename:
```php
$this->file->getClientOriginalName();
```

---
https://laravel-livewire.com/docs/2.x/file-uploads#storing-files

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)

**Comment by JulioPablo** [comment]
@nuernbergerA did you even read the issue description? 

Your comment doesn't really have much to do with what I'm describing.
Basically calling `$file->getPath()` or `$file->getFilename()` to get the **current** path or filename, which in default-configured liverwire instance would be located at `livewire-tmp/` are returning a completely different path & filename.

**Comment by nuernbergerA** [comment]
I did read your issue.

Did you read the docs on how to store files?

You don't have to call those methods.

At this stage, the file is only stored in a `tmp` folder, if you wanna keep the uploaded file and rename it, you need to `store` it.

So if you upload 'cv.docx' and you want to store it but get the extension use `->getClientOriginalName();` to get `cv.docx`

Maybe you can post a full code example to illustrate what you want to achieve instead of saying whats not possible.

Cheers

**Comment by JulioPablo** [comment]
Sure did, the issue is not related with storing files, that works fine, but the results from calling `$file->getPath()` or `$file->getFilename()`  on a `TemporaryUploadedFile` are plain wrong. 

If that's ok with you as maintainers of the project that's fine by me, I just thought I'd report it.

**Comment by amiranagram** [comment]
@JulioPablo `getPath` is fixed in #4001 and tagged in `2.7.0`. You can get the original filename with `$file->getClientOriginalName()`.
